app/controllers/concerns/foreman/controller/taxonomies_controller.rb

```diff
@@ -135,19 +135,19 @@ def import_mismatches
 
   def assign_hosts
     @taxonomy_type = taxonomy_single.classify
-    @hosts = Host.authorized(:view_hosts, Host).send("no_#{taxonomy_single}").includes(included_associations).search_for(params[:search],:order => params[:order]).paginate(:page => params[:page])
+    @hosts = hosts_scope.send("no_#{taxonomy_single}").includes(included_associations).search_for(params[:search],:order => params[:order]).paginate(:page => params[:page])
     render "hosts/assign_hosts"
   end
 
   def assign_all_hosts
-    Host.send("no_#{taxonomy_single}").update_all(taxonomy_id => @taxonomy.id)
+    hosts_scope.send("no_#{taxonomy_single}").update_all(taxonomy_id => @taxonomy.id)
     @taxonomy.import_missing_ids
     redirect_to send("#{taxonomies_plural}_path"), :notice => _("All hosts previously with no %{single} are now assigned to %{name}") % { :single => taxonomy_single, :name => @taxonomy.name }
   end
 
   def assign_selected_hosts
     host_ids = params[taxonomy_single.to_sym][:host_ids] - ["0"]
-    @hosts = Host.where(:id => host_ids).update_all(taxonomy_id => @taxonomy.id)
+    @hosts = hosts_scope.where(:id => host_ids).update_all(taxonomy_id => @taxonomy.id)
```

Comment:
`hosts_scope.send("no_#{taxonomy_single}")` was missing here, so the manual selection was not working properly. I would be ok fixing this as part of this PR, as the whole assignment was just broken.